engine/include/cubos/engine/collisions/collider.hpp

```diff
@@ -0,0 +1,29 @@
+/// @file
+/// @brief Component @ref cubos::engine::Collider.
+/// @ingroup collisions-plugin
+
+#pragma once
+
+#include <glm/mat4x4.hpp>
+
+#include <cubos/engine/collisions/aabb.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Component which adds a collider to an entity.
+    /// @ingroup collisions-plugin
+    struct [[cubos::component("cubos/collider", VecStorage)]] Collider
+    {
+        glm::mat4 transform{1.0F}; ///< Transform of the collider.
+
+        ColliderAABB localAABB; ///< Local space AABB of the collider.
+        ColliderAABB worldAABB; ///< World space AABB of the collider.
+
+        /// @brief Margin of the collider.
+        ///
+        /// When the collider shape has sharp edges, a margin is needed.
+        float margin;
```

Comment:
Shouldn't we provide a sane default? Otherwise the user will have to specify it or just leave it to the will of the UB god.

Reply:
I'm setting it in the plugin, based on the associated shape. It isn't meant to be modified by users.

Reply:
Then it should be documented as such.

Reply:
done